kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/SessionTrackerCookie.java

```diff
@@ -39,17 +39,21 @@ private SessionTrackerCookie() {
      *            the HTTP response.
      */
     public static void setIfNeeded(HttpSession session,
-            HttpServletRequest request, HttpServletResponse response) {
+            HttpServletRequest request, HttpServletResponse response,
+            SameSite sameSite) {
         Optional<Cookie> clusterKeyCookie = getCookie(request);
-        if (!clusterKeyCookie.isPresent()) {
+        if (clusterKeyCookie.isEmpty()) {
             String clusterKey = UUID.randomUUID().toString();
             session.setAttribute(CurrentKey.COOKIE_NAME, clusterKey);
-            response.addCookie(new Cookie(CurrentKey.COOKIE_NAME, clusterKey));
+            Cookie cookie = new Cookie(CurrentKey.COOKIE_NAME, clusterKey);
+            cookie.setHttpOnly(true);
+            cookie.setPath(request.getContextPath());
```

Comment:
I suppose this needs some additional computation. For root context, `getContextPath()` will return ''. Perhaps we need to set it on the cookie as '/'